### PR TITLE
Support exist condition for metric alert

### PR DESCRIPTION
**Pull request by aiwantaozi**
Problem:
Now we only support compare condition and the threshold for metric alert

Solution:
Add exist condition for metric alert

Issue:
https://github.com/rancher/rancher/issues/21302

Types:
https://github.com/rancher/types/pull/914

### CODE REVIEW

**Comment by aiwantaozi** [comment]
Rebased

File: pkg/controllers/user/alert/deployer/notification_template.go [hunk 1]
```diff
@@ -105,8 +105,12 @@ Project Name: {{ .Labels.project_name}}{{ end }}
 {{- if .Labels.pod_name }}
 Pod Name: {{ .Labels.pod_name}}{{ else if .Labels.pod -}}Pod Name: {{ .Labels.pod}}{{ end }}
 Expression: {{ .Labels.expression}}
+{{- if .Labels.threshold_value }}
+Description: The configured event happened for ({{ .Labels.duration}}): expression matched, datapoint value is {{ .Annotations.current_value}}
+{{- else}}
```
Comment (gitlawr) [inline pkg/controllers/user/alert/deployer/notification_template.go]: Looks like we print `Threshold Crossed ...` when `threshold_value` is empty. Should we do it the other way?

**Comment by aiwantaozi** [comment]
Grafana has an alert for is null value for the expression result, if we use null would be a negative option like ```not null value```, so I change to opposite option ```has-value```

**Comment by gitlawr** [comment]
LGTM

**Comment by luthermonson** [comment]
LGTM, needs conflict resolved though

**Comment by cjellick** [comment]
needs rebase but lgtm

**Comment by aiwantaozi** [comment]
rebased, thanks

**Pull request merged**